0021-zarf-values/README.md

```diff
@@ -0,0 +1,291 @@
+
+This feature also could open up Zarf packages to being less declarative - especially if a package author opens up security-critical Helm values in their charts.  This caveat should be clearly documented as a concern which should also recommend a policy engine be used to enforce security-critical values within the cluster itself.
+
+Because we will be using more `interface{}` types, we should also look into the security implications of this feature and ensure that this is well tested and we utilize some of Helm's existing protections against `nil` maps and other potential security issues with this feature.
+
+## Design Details
+
+The new `values.files` field would be added to the `ZarfPackageConfig` schema and would accept a list of local values files or URLs, matching the existing functionality of the `valuesFiles` key under `charts`.  This key would also follow the same composability logic as the `valuesFiles` key with additional parent values files merging with and overriding any common keys from children imports.  Since this is a global field, values would be merged regardless of component similar to how variables work today.  The `values.schema` field would simply replace the child version if it were non-empty in the parent, similar to the `namespace` or `releaseName` fields in `charts` today.
+
+Each of the referenced values files would be included inside the created Zarf package and stored inside of the tarball or at an OCI path so that they would travel with the package and be available to the user on deploy.  On deploy, the values (from the defaults in the `ZarfPackageConfig` and the set values files) would be merged with the set values overriding the defaults, and the resulting values would be passed to the Helm chart via the `values` field under a given `charts` entry.  Any actions which contained a go template (i.e. `{{ .Values.component.resources | toYaml }}`) would be templated prior to execution. This templating would work in all fields within an action definition including `cmd` and `wait` actions.
+
+This feature would also implement a `setValues` field for actions that would act as a replacement for the existing `setVariables` field.  This field would take a values path and would set that path to the output of the command.  This field would also have `type` defined on it, though instead of `file`, it would take `string`, `json`, or `yaml` and then handle the standard out of the command according to that format.
+
+Zarf Values would also be available to onCreate and onRemove actions if they existed in `package.create.values` or `package.remove.values` or were set with `-f`/`--values`.  These would template actions and be able to use `setValues` but would not map to Helm charts.
```

Comment:
While it would not be a new feature, giving users another avenue to run arbitrary bash commands based on deploy time input on the host does worry me a bit. It would be good to get some examples in the motivation so we can see why a user might want this feature.

Reply:
I'll add references to examples I know of - though also welcome to bash!
![image](https://github.com/user-attachments/assets/9144db55-0ead-4210-b5f6-2f429aca8fb6)

Reply:
added a note on this - though will also want to think on this a bit since there are a few other ideas to think about here.  this isn't too different from other controlled input runner tools (i.e GitHub/GitLab actions) and generally we expect a user who has access to Zarf to be an administrator though it does need to be clearly documented how to mitigate injection possibilities.

Reply:
yep - and if you notice that action in the gitlab runner is actually running that last command in the k8s cluster (https://github.com/defenseunicorns/uds-package-gitlab-runner/blob/d2b573bdbed12ac2aafd52082f1b9ea84b213439/common/zarf.yaml#L42) and is otherwise only getting basic k8s information.  The general assumption we have pushed in UDS Packages is this: 

![image](https://github.com/user-attachments/assets/3f54c924-b35a-47f9-9e5c-11a80311f030)
https://github.com/defenseunicorns/uds-common/blob/main/docs/uds-packages/requirements/uds-package-requirements.md

That is likely something we could expand on our side though we'd want to make sure the community is ready to come with us on that (potentially in another ZEP)

Reply:
For our own UDS Packages we have pushed them to only depend on `./zarf` and basic coreutils stuff like `if` - if there are other dependencies those need to run in the cluster since we don't know where those would be on the deploying system and we need to bring them with us (which is easier to do as a container in the cluster)